app/src/main/java/com/reminder/hydration/SharedPrefManager.kt

```diff
@@ -0,0 +1,52 @@
+package com.reminder.hydration
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class SharedPrefManager private constructor(context: Context) {
+    private val sharedPreferences: SharedPreferences =
+        context.getSharedPreferences("Hydration", Context.MODE_PRIVATE)
+
+
+    fun <T> save(key: String, value: T) {
+        when (value) {
+            is String -> {
+                sharedPreferences.edit()
```

Comment:
You are repeating this line 4 times. Do it when function gets called and put it in a variable and use it.